x-pack/plugins/security_solution/public/common/components/filter_group/index.tsx

```diff
@@ -231,40 +234,28 @@ const FilterGroupComponent = (props: PropsWithChildren<FilterGroupProps>) => {
 
     // if initialUrlParam Exists... replace localInitialControls with what was provided in the Url
     if (overridingControls && !urlDataApplied.current) {
-      let maxInitialControlIdx = Math.max(
-        localInitialControls.length - 1,
-        (overridingControls?.length ?? 1) - 1
-      );
-      for (let counter = overridingControls.length - 1; counter >= 0; counter--) {
-        const urlControl = overridingControls[counter];
-        const idx = localInitialControls.findIndex(
-          (item) => item.fieldName === urlControl.fieldName
-        );
-
-        if (idx !== -1) {
-          // if index found, replace that with what was provided in the Url
-          resultControls[idx] = {
-            ...localInitialControls[idx],
-            fieldName: urlControl.fieldName,
-            title: urlControl.title ?? urlControl.fieldName,
-            selectedOptions: urlControl.selectedOptions ?? [],
-            existsSelected: urlControl.existsSelected ?? false,
-            exclude: urlControl.exclude ?? false,
-          };
-        } else {
-          // if url param is not available in initialControl, start replacing the last slot in the
-          // initial Control with the last `not found` element in the Url Param
-          //
-          resultControls[maxInitialControlIdx] = {
-            fieldName: urlControl.fieldName,
-            selectedOptions: urlControl.selectedOptions ?? [],
-            title: urlControl.title ?? urlControl.fieldName,
-            existsSelected: urlControl.existsSelected ?? false,
-            exclude: urlControl.exclude ?? false,
-          };
-          maxInitialControlIdx--;
-        }
+      if (localInitialControls.length > 0) {
+        localInitialControls.forEach((persistControl) => {
+          const doesPersistControlAlreadyExist = overridingControls?.findIndex(
```

Comment:
Super nit, but you could use `Array.some` to get a boolean response so that `doesPersistControlAlreadyExist` is a boolean since we don't do anything with the index